tsdb/engine/tsm1/engine.go

```diff
@@ -882,7 +884,7 @@ func (e *Engine) LoadMetadataIndex(shardID uint64, index tsdb.Index) error {
 	}
 
 	// Save the field set index so we don't have to rebuild it next time
-	if err := e.fieldset.Save(); err != nil {
+	if err := e.fieldset.Save(e.MeasurementFieldSet().GetAndClearChanges()); err != nil {
```

Comment:
Is it bad if `e.fieldset.Save` fails but we've already cleared the set of changed fields? Do we need to "unget" the changes if `.Save` fails?

Reply:
I think my code is actually wrong here; rather than saving incremental changes, we should save the whole shebang here.